classes/core/traits/ModelWithSettings.php

```diff
@@ -183,4 +197,35 @@ protected function id(): Attribute
             set: fn ($value) => [$this->primaryKey => $value],
         );
     }
+
+    /**
+     * @see \Illuminate\Database\Eloquent\Concerns\GuardsAttributes::isGuardableColumn()
+     */
+    protected function isGuardableColumn($key)
+    {
+        // Need the snake like to key to check for main table to compare with column listing
+        $key = Str::snake($key);
+        
+        if (! isset(static::$guardableColumns[get_class($this)])) {
+            $columns = $this->getConnection()
+                        ->getSchemaBuilder()
+                        ->getColumnListing($this->getTable());
+
+            if (empty($columns)) {
+                return true;
+            }
+            static::$guardableColumns[get_class($this)] = $columns;
+        }
+
+
+        $settingsWithMultilingual = array_merge($this->getSettings(), $this->getMultilingualProps());
+        $camelKey = Str::camel($key);
+        
+        // Check if this column included in setting and multilingula props and not set to guarded
+        if (in_array($camelKey, $settingsWithMultilingual) && !in_array($camelKey, $this->getGuarded())) {
+            return true;
+        }
+        
+        return in_array($key, (array)static::$guardableColumns[get_class($this)]);
+    }
```

Comment:
@Vitaliy-1 this override of `\Illuminate\Database\Eloquent\Concerns\GuardsAttributes::isGuardableColumn` seems required when defining any guarded attributes for entities with no schema map as for schema mapped entities, we always generate the `fillable` .